src/data/store.ts

```diff
@@ -61,38 +55,53 @@ export const query: Writable<Query> = writable({
     query: ""
 });
 
+export const allSongs: Writable<Song[]> = writable([]);
+export const queriedSongs: Writable<Song[]> = writable([]);
+
 export const isPlaying = writable(false);
-export const currentSong: Writable<Song> = writable(null);
-export const currentSongIdx = writable(0);
+export const current: Writable<PlayingSong> = writable({ song: null, index: 0, position: 0 }, async () => {
+    const item = localStorage.getItem("current");
 
-const defaultLastPlayedInfo: LastPlayedInfo = {
-    songId: null,
-    position: 0
-};
+    if (item) {
+        const data = JSON.parse(item);
 
-export const lastPlayedInfo: Writable<LastPlayedInfo> = writable(
-    {
-        ...defaultLastPlayedInfo,
-        ...JSON.parse(localStorage.getItem("lastPlayedInfo"))
-    } || defaultLastPlayedInfo
-);
-// Auto-persist settings
-lastPlayedInfo.subscribe((val) =>
-    localStorage.setItem("lastPlayedInfo", JSON.stringify(val))
-);
+        if (data.song) {
+            const song = await db.songs.get(data.song);
 
-export const nextUpSong: Writable<Song> = writable(null);
-export const queriedSongs: Writable<Song[]> = writable([]);
+            current.set({ song, index: data.index, position: data.position });
 
-export const allSongs: Writable<Song[]> = writable([]);
-export const playlist: Writable<Song[]> = writable([]);
-export const shuffledPlaylist: Writable<Song[]> = writable([]);
-export const albumPlaylist: Writable<Song[]> = writable([]);
-export const playlistIsAlbum = writable(false);
-export const playlistCountry = writable(null); // ISO Country code
-export const playlistType: Writable<PlaylistType> = writable("library");
-export const playlistDuration: Writable<number> = writable(0);
+            playerTime.set(data.position);
+        }
+    }
+});
+current.subscribe(({ song, index, position }) => {
+    const data = song
+        ? { song: song.id, index, position }
+        : { song: null, index: 0, position: 0 };
+
+    localStorage.setItem("current", JSON.stringify(data));
+});
+export const queue: Writable<Song[]> = writable([], async () => {
+    const item = localStorage.getItem("queue");
```

Comment:
Persisting is great, but perhaps we shouldn't write the queue to localStorage since [browsers limit the size to 5MB](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria). Music hoarders with huge libraries would easily reach this limitation (for my library of 1200 songs this value is 37 KB, so 120,000 songs would be approx 3.7 MB). 

We could either write it to a blob in IndexedDB, or just use a regular queue.txt file in the config directory. I think I prefer the file option, it's simple and the contents of the file would be the same as you have now.

Reply:
Oh! Yay, I knew I forgot something when writing the PR...

I thinks there is 2 options:
- limit the size (100 or 1000 from the first played song) when playing from the library.
  I'm afraid that a string will slow down the startup (I do a split to get back the array).
- have a "library" mode where you:
  - can select a song to play
  - can navigate
  - can't modify the order
  - can't add/delete songs

Reply:
Modes are too confusing, and if we can persist the whole list, why not? I did some testing with the file approach on my M1 MBP:
- 1200 songs (37 KB file) - 350ms to restore the queue. Barely noticeable on app startup. 
- 120,000 songs (3.7 MB file) - 4600ms to restore the queue. 
I think that's fine, it doesn't have to be instant. I'll explore other options in the future but if you don't mind I can push to this PR?

Reply:
I don't mind at all. You can go ahead with the merge.